### ISSUE TITLE
AhoyEmail.secret_token is nil causing TypeError: no implicit conversion of nil into String

### ISSUE THREAD

**Issue by rahilsondhi** [ankane/ahoy_email#15]
Hey Andrew,

I installed ahoy_email in my rails 4.1.5 app. When I tried to send an email, I got this error:

```
TypeError: no implicit conversion of nil into String
from /Users/rahil/.rvm/gems/ruby-2.1.1/gems/ahoy_email-0.2.1/lib/ahoy_email/processor.rb:106:in `hexdigest'
```

In rails console, `AhoyEmail.secret_token` returns nil.

I generated a secret with `rake secret` and tried to put it in an initializer like `AhoyEmail.secret_token = SECRET` or `AhoyEmail.options[:secret_token] = SECRET` but neither worked.

Could this have something to do with the [new rails 4.1 secrets.yml file](http://edgeguides.rubyonrails.org/4_1_release_notes.html#config-secrets-yml)?

Any tips? Thanks!

**Comment by yagudaev** [comment]
Has this been fixed in 0.2.2? If not, can we get a new version of the gem published?

**Comment by ankane** [comment]
This is fixed in 0.2.2.

**Comment by yagudaev** [comment]
Yeah thanks just needed to set the secret in the console and make sure we were using the new secrets.yml in our rails 4.1 projects (older project previously upgraded to newest rails).
